plugin/trino-hive/src/main/java/io/trino/plugin/hive/ion/IonFileWriterFactory.java

```diff
@@ -95,12 +100,17 @@ public Optional<FileWriter> createFileWriter(
                     .mapToObj(ordinal -> new Column(fileColumnNames.get(ordinal), fileColumnTypes.get(ordinal), ordinal))
                     .toList();
 
+            IonWriterBuilder writerBuilder = getIonEncoding(schema).equals(TEXT_ENCODING)
```

Comment:
Prefer to fail on a malformed or unsupported encoding option. e.g. in future we may want to support `JSON` but we don't today so that should hard fail; it would be surprising if instead it wrote binary ion.

Reply:
I agree, should have the case to handle the unsupported option.